src/hyperloglog.c

```diff
@@ -1350,12 +1580,16 @@ void pfmergeCommand(client *c) {
 
     /* Write the resulting HLL to the destination HLL registers and
      * invalidate the cached value. */
-    for (j = 0; j < HLL_REGISTERS; j++) {
-        if (max[j] == 0) continue;
-        hdr = o->ptr;
-        switch(hdr->encoding) {
-        case HLL_DENSE: hllDenseSet(hdr->registers,j,max[j]); break;
-        case HLL_SPARSE: hllSparseSet(o,j,max[j]); break;
+    if(use_dense) {
+        hllDenseCompress(hdr->registers, max);
+    } else {
+        for (j = 0; j < HLL_REGISTERS; j++) {
+            if (max[j] == 0) continue;
+            hdr = o->ptr;
+            switch (hdr->encoding) {
+                case HLL_DENSE: hllDenseSet(hdr->registers,j,max[j]); break;
+                case HLL_SPARSE: hllSparseSet(o,j,max[j]); break;
+            }
```

Comment:
what about jumping out in advance when using hllDenseCompress(), in which way we can avoid touching the original code.
like
```
if(use_dense) {
    hllDenseCompress(hdr->registers, max);
   goto end;  
}
for (j = 0; j < HLL_REGISTERS; j++) {
```

Reply:
Personally I think it seems weird to replace a simple `if-else` with `goto`.
But I can change the code here if keeping the original code is more important.
Which one do you prefer?

Reply:
i'm hesitated for it, but it doesn't change too much, we can leave it as it is.

Reply:
I prefer current code, i don't think it is a good idea to avoid touching the original code but make it less readable.